src/Durable/DurableTimer.cs

```diff
@@ -0,0 +1,79 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+namespace Microsoft.Azure.Functions.PowerShellWorker.Durable
+{
+    using System;
+    using System.Linq;
+    using System.Threading;
+
+    internal class DurableTimer
+    {
+        private readonly ManualResetEvent _waitForStop = new ManualResetEvent(initialState: false);
+
+        public void CreateTimerAndStop_OrContinue(OrchestrationContext context, DateTime fireAt)
+        {
+            context.OrchestrationActionCollector.Add(new CreateDurableTimerAction(fireAt));
+
+            var timerCreated = GetTimerCreated(context, fireAt);
+            var timerFired = GetTimerFired(context, timerCreated);
+
+            if (timerCreated == null)
+            {
+                CreateTimerAndWaitUntilFired(context);
+            }
+            else if (timerFired != null)
+            {
+                // Update CurrentUtcDateTime if the timer has fired
+                var newOrchestrationStart = context.History.FirstOrDefault(
+                        e => e.EventType == HistoryEventType.OrchestratorStarted &&
+                            !e.IsProcessed
+                    );
+                
+                if (newOrchestrationStart != null)
+                {
+                    context.CurrentUtcDateTime = newOrchestrationStart.Timestamp.ToUniversalTime();
+                    newOrchestrationStart.IsProcessed = true;
+                }
```

Comment:
We have this piece of code (lines 30-39) duplicated multiple times now (once here and twice in `ActivityInvocationTracker`). I think it deserves to be extracted to a method, perhaps on a new class (something like `CurrentUtcDateTimeUpdater`).